[JLIFE-8] feat: SettingView 생성 및 Main에서 모달 present

iveriie opened the pull request:
- Setting View Modal 구현 확인을 위한 임시 코드에 주석 temporary code 표시

iveriie committed:
[JLIFE-8] feat: SettingView 생성 및 Main에서 모달 present

- Setting View Modal 구현 확인을 위한 임시 코드에 주석 temporary code 표시

serena0720 commented on JLife/JLife/MainViewController.swift:
```diff
@@ -99,6 +105,16 @@ final class MainViewController: UIViewController {
+    @objc
+    private func presentModalSettingViewController() {
+        let settingVC = SettingViewController()
+        settingVC.modalTransitionStyle = .coverVertical
+        settingVC.modalPresentationStyle = .automatic
```
default 값 확인해서 불필요한 라인 삭제해주세요~

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+
+final class SettingView: UIView {
+
+    private let assetsNameList = ["Basic", "Beige", "Blue", "CoolGray", "Coral", "Green", "Pink", "Purple", "WarmGray", "Yellow"]
```
별도의 객체로 분리하면 어떨까요?

serena0720 replied:
```swift
enum ColorCase {
  case basic
  case blue

  var theme: (Sting: String: Sting) {
    switch self {
      case .basic:
              // code
      case .blue:
              // code
    }
  }
}
```
이런식으로 계산 프로퍼티로 활용할 수도 있어용

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+        label.text = "달력 색상 변경"
+        label.textAlignment = .center
+        label.font = UIFontMetrics(forTextStyle: .title2).scaledFont(for: customFont!)
+        label.adjustsFontForContentSizeCategory = true
```
통일성있게 pickerView내부 Label에도 적용해주세요

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+    
+    private let nowThemeLabel: UILabel = {
+        let label = UILabel()
+        let customFont =  UIFont(name: "Cafe24Ssurroundair", size: 17)
```
사이즈 17로 지정하지말고 system size를 사용하는게 동적 대응이돼서 좋을 거 같아용

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+        label.text = "현재 테마 색상: (-)" // 현재 테마 받아와서 넣어야함 / 기존방식: userdefaults
+        label.textAlignment = .center
+        label.adjustsFontForContentSizeCategory = true
+        label.font = UIFontMetrics(forTextStyle: .body).scaledFont(for: customFont!)
```
강제 언래핑말고 default값을 주는 쪽으로 언래핑하는게 좋을 거 같아용

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+        let button = UIButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setTitle("적용", for: .normal)
+        button.titleLabel?.font = UIFont(name: "Cafe24Ssurroundair", size: 17)
```
사이즈 systemfont사이즈로 통일되게 수정해주세용

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+        return button
+    }()
+    
+    // MARK: - Intializer
```
해당 주석 삭제해주세용

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+        configureUI()
+        setUpAllConstraints()
+        
+        themePickerView.delegate = self
+        themePickerView.dataSource = self
+        
+        applyButton.addTarget(self, action: #selector(touchApplyButton), for: .touchUpInside)
```
두 개의 함수로 분리해서 depth 맞춰주세용

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    @objc
```
// MARK: - Private Funtion으로 통일해주세용

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+    }
+    
+    // MARK: - Configure UI
+    private func setUpBackgroundColor() {
```
configureBackgroundColor로 통일해주세용

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+    private func setUpTitleLabelConstraints() {
+        NSLayoutConstraint.activate([
+            titleLabel.centerXAnchor.constraint(equalTo: centerXAnchor),
+            titleLabel.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor, constant: 30),
```
전체 component를 하나의 StackView에 넣고 StackView를 화면 크기에 맞춘 후, 그 안에서 비율로 constant를 잡아주세용

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+    func pickerView(_ pickerView: UIPickerView, viewForRow row: Int, forComponent component: Int, reusing view: UIView?) -> UIView {
+            var label = UILabel()
+            if let labelView = view {
+                label = labelView as! UILabel
```
강제 언래핑 말고 default값으로 맞춰주세용

serena0720 commented on JLife/JLife/SettingView.swift:
```diff
@@ -0,0 +1,198 @@
+                label = labelView as! UILabel
+            }
+            label.font = UIFont(name: "Cafe24Ssurroundair", size: 17)
+            label.text = assetsNameList[row] // 한글로 변환 해야할지?
```
enum으로 분리하여 CaseIterable채택하는쪽으로 수정해주세여
> https://stackoverflow.com/questions/27094878/how-do-i-get-the-count-of-a-swift-enum

serena0720 commented on JLife/JLife/SettingViewController.swift:
```diff
@@ -9,20 +9,31 @@ import UIKit
+    // MARK: - ViewDidLoad
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        NotificationCenter.default.addObserver(self,
```
동일하게 depth 맞춰주세용

serena0720 commented on JLife/JLife/SettingViewController.swift:
```diff
@@ -45,7 +56,14 @@ final class SettingViewController: UIViewController {
+    @objc
+    private func popToMainViewControllerWithApplyTheme() {
+        
+        // 적용버튼을 누른거니까 색상 테마를 여기서 적용해야함
```
불필요한 주석들은 모두 삭제 해주세용

serena0720 commented:
@monargent0 너무 고생했어👍
잘한다 잘한다🙌

iveriie committed:
[JLIFE-8] refactor: SettingView 전반 코드수정, CalendarColorPalette 파일 생성

- assets BasicTheme, BasicBorder 색상명 변경
- SettingViewController 이전 코드 삭제
- 강제추출, 수치 제거 및 수정
- Setting View의 Components 전체 스택뷰로 수정
- 색상명 enum 객체 분리 파일명 CalendarColorPalette

monargent0 commented:
@serena0720 수정했습니다
